### ISSUE TITLE
Wrangler is reinstalled even when it's already available

### ISSUE THREAD

**Issue by 1000hz** [cloudflare/wrangler-action#199]
Currently, `wrangler-action` always installs wrangler as specified by the `wranglerVersion` input or the `DEFAULT_WRANGLER_VERSION` constant contained in the source. This isn't always necessary as Wrangler may have already been installed in a previous step of the workflow (e.g. if it's declared in the project's dependencies in `package.json`). This also makes it harder to keep the version of Wrangler used by `wrangler-action` and the version used locally consistent, as the user needs to manually keep `wranglerVersion` and `package.json` in sync.

Instead, we should only install Wrangler if it's not already present in the `$PATH` (including the package manager bin directory) OR if `wranglerVersion` has been explicitly provided

**Comment by peplin** [comment]
+1 but also wanted to mention that not having the option to disabling this install is preventing my team from using this action because of the hugely increased runtime. The action runs `yarn add wrangler` and triggers linking for our entire (large) list of dependencies, and this action takes > 2 minutes. `npx wrangler` is ready in < 3 seconds, for comparison.

**Comment by AdiRishi** [comment]
I've recently moved all my projects to use `wrangler-action` in CI. Definitely hit this quite a few times now.
I've made a PR with a fix 🙏

**Comment by TonyRL** [comment]
@petebacondarwin Please reopen this issue due to #265.

**Comment by wqcstrong** [comment]
> I've recently moved all my projects to use `wrangler-action` in CI. Definitely hit this quite a few times now. I've made a PR with a fix 👍

@AdiRishi Thanks for your contribution. However, I am currently facing some issues with my project setup. My project uses yarn workspaces. Even though I have executed `yarn install` which installs wrangler@3.61.0 at the workspace root in `bash ./build.sh` , it seems that the check here https://github.com/cloudflare/wrangler-action/blob/main/src/index.ts#L117-L124 did not pass. 

```json
// "package.json" in workspace root 
{
  "name": "sony-pages",
  "workspaces": [
    "apps/*"
  ],
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "deploy": "wrangler deploy",
    "dev": "wrangler dev",
    "start": "wrangler dev",
    "test": "vitest"
  },
  "devDependencies": {
    "@cloudflare/kv-asset-handler": "^0.3.3",
    "@cloudflare/workers-types": "^4.20240620.0",
    "lerna": "^8.1.5",
    "typescript": "^5.0.4",
    "vitest": "1.3.0",
    "wrangler": "^3.61.0"
  },
  "dependencies": {}
}
```

```yaml
// the workflow yml
name: Deploy

on:
  push:
    branches:
      - main

jobs:
  deploy:
    runs-on: ubuntu-latest
    name: Deploy
    steps:
      - uses: actions/checkout@v4

      - name: Build
        run: |
          bash ./build.sh

      - name: Deploy to cloudflare
        uses: cloudflare/wrangler-action@v3.6.0
        with:
          apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}

      - name: Lark bot notify
        run: |
          curl -X POST \
              -H "Content-Type: application/json" \
              ...
```

Do you know what could be the reason for this? Below is a screenshot of the results from my workflow execution.

![image](https://github.com/cloudflare/wrangler-action/assets/23474513/89b8b305-8566-4a6d-bac1-e69bab1f967d)

**Comment by wqcstrong** [comment]
> > I've recently moved all my projects to use `wrangler-action` in CI. Definitely hit this quite a few times now. I've made a PR with a fix 👍
> 
> @AdiRishi Thanks for your contribution. However, I am currently facing some issues with my project setup. My project uses yarn workspaces. Even though I have executed `yarn install` which installs wrangler@3.61.0 at the workspace root in `bash ./build.sh` , it seems that the check here https://github.com/cloudflare/wrangler-action/blob/main/src/index.ts#L117-L124 did not pass.
> 
> ```json
> // "package.json" in workspace root 
> {
>   "name": "sony-pages",
>   "workspaces": [
>     "apps/*"
>   ],
>   "version": "0.0.0",
>   "private": true,
>   "scripts": {
>     "deploy": "wrangler deploy",
>     "dev": "wrangler dev",
>     "start": "wrangler dev",
>     "test": "vitest"
>   },
>   "devDependencies": {
>     "@cloudflare/kv-asset-handler": "^0.3.3",
>     "@cloudflare/workers-types": "^4.20240620.0",
>     "lerna": "^8.1.5",
>     "typescript": "^5.0.4",
>     "vitest": "1.3.0",
>     "wrangler": "^3.61.0"
>   },
>   "dependencies": {}
> }
> ```
> 
> ```yaml
> // the workflow yml
> name: Deploy
> 
> on:
>   push:
>     branches:
>       - main
> 
> jobs:
>   deploy:
>     runs-on: ubuntu-latest
>     name: Deploy
>     steps:
>       - uses: actions/checkout@v4
> 
>       - name: Build
>         run: |
>           bash ./build.sh
> 
>       - name: Deploy to cloudflare
>         uses: cloudflare/wrangler-action@v3.6.0
>         with:
>           apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
> 
>       - name: Lark bot notify
>         run: |
>           curl -X POST \
>               -H "Content-Type: application/json" \
>               ...
> ```
> 
> Do you know what could be the reason for this? Below is a screenshot of the results from my workflow execution.
> 
> ![image](https://private-user-images.githubusercontent.com/23474513/342732188-89b8b305-8566-4a6d-bac1-e69bab1f967d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk1NTQyOTUsIm5iZiI6MTcxOTU1Mzk5NSwicGF0aCI6Ii8yMzQ3NDUxMy8zNDI3MzIxODgtODliOGIzMDUtODU2Ni00YTZkLWJhYzEtZTY5YmFiMWY5NjdkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI4VDA1NTMxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYxMDBjMGUwMWMzNzUxODg3Yjk4ZTRiMjhmYTU0MmY3Y2E1OTg5ZDkzYWJkNGQwMDA2NzUyN2Q3NzNmZmM4ZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.eyhGUhCTy1venqmpvE1SZ1jExA_RCVBWz3WyaXOSQao)

Delete 'workspaces' field in package.json is one solution:

```diff
name: Deploy

on:
  push:
    branches:
      - main

jobs:
  deploy:
    runs-on: ubuntu-latest
    name: Deploy
    steps:
      - uses: actions/checkout@v4

      - name: Build
        run: |
          bash ./build.sh

+      - name: Delete 'workspaces' field
+        run: |
+          jq 'del(.workspaces)' package.json > temp.json
+          mv temp.json package.json
+          cat package.json

      - name: Deploy to cloudflare
        uses: cloudflare/wrangler-action@v3
        with:
          apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}

      - name: Lark bot notify
        run: |
          curl -X POST \
              -H "Content-Type: application/json" \
              .......
```

**Comment by tyteen4a03** [comment]
Also running into this issue.

**Comment by acusti** [comment]
@wqcstrong i think you are running into the issue i documented in #277. for now, i just pinned the version in my workflow file (`wranglerVersion: 3.64.0`), but hopefully the fix will be merged soon (cc: @tyteen4a03 in case you have the same issue)

**Comment by adamalfredsson** [comment]
Same issue :(

> Run cloudflare/wrangler-action@v3.7
> 🔍 Checking for existing Wrangler installation
>   /usr/local/bin/yarn wrangler --version
>   yarn run v1.22.22
>   $ /home/runner/work/project/node_modules/.bin/wrangler --version
>   3.13.2
>   Done in 0.60s.
>   ⚠️ Wrangler not found or version is incompatible. Installing...
> 📥 Installing Wrangler
>   /usr/local/bin/yarn add wrangler@3.13.2
>   yarn add v1.22.22
>   error Running this command will add the dependency to the workspace root rather than the workspace itself, which might not be what you want - if you really meant it, make it explicit by running this command again with the -W flag (or --ignore-workspace-root-check).
>   info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
> Error: The process '/usr/local/bin/yarn' failed with exit code 1

**Comment by flyck** [comment]
Ran into an issue related to this for a project with [honox](https://github.com/honojs/honox)

Since an old version is installed, the vite build fails with a commonJS error. This error seems to be related to an [issue](https://github.com/cloudflare/workers-sdk/issues/5311) with this older version of wrangler, and I could resolve it by specifying a current wrangler version (3.80.0):
```sh
  bun run build
  $ vite build --mode client && vite build
  failed to load config from /home/runner/work/RateThatIssue/RateThatIssue/vite.config.ts
  error during build:
  file:///home/runner/work/RateThatIssue/RateThatIssue/node_modules/@hono/vite-dev-server/dist/adapter/cloudflare.js:2
  import { getPlatformProxy } from "wrangler";
           ^^^^^^^^^^^^^^^^
  SyntaxError: Named export 'getPlatformProxy' not found. The requested module 'wrangler' is a CommonJS module, which may not support all module.exports as named exports.
  CommonJS modules can always be imported via the default export, for example using:
  
  import pkg from 'wrangler';
  const { getPlatformProxy } = pkg;
```

**Comment by austinorth** [comment]
Same issue here.
```
  /usr/local/bin/yarn wrangler --version
  yarn run v1.22.22
  $ /home/runner/work/customer-frontend/customer-frontend/node_modules/.bin/wrangler --version
  3.78.3
  Done in 0.89s.
  ⚠️ Wrangler not found or version is incompatible. Installing...
📥 Installing Wrangler
  /usr/local/bin/yarn add wrangler@3.13.2
```

**Comment by austinorth** [comment]
Same issue here.
```
  /usr/local/bin/yarn wrangler --version
  yarn run v1.22.22
  $ /home/runner/work/X/X/node_modules/.bin/wrangler --version
  3.78.3
  Done in 0.89s.
  ⚠️ Wrangler not found or version is incompatible. Installing...
📥 Installing Wrangler
  /usr/local/bin/yarn add wrangler@3.13.2
```

**Comment by jahands** [comment]
This should be fixed now via #278

If anyone is still having this issue, please let us know.
Thanks!
